### ISSUE TITLE
og:image and title

### ISSUE THREAD

**Issue by mirik999** [nygardk/react-share#149]
Hi people, how to correct set sharing ? 
i attach 3 screenshot , maybe i did mistake.
![screenshot_4](https://user-images.githubusercontent.com/29675733/38669859-604ad1bc-3e58-11e8-867d-4b2f64e4699a.png)
![screenshot_2](https://user-images.githubusercontent.com/29675733/38669864-65aa18f2-3e58-11e8-8168-120710fa381c.png)
![screenshot_3](https://user-images.githubusercontent.com/29675733/38669876-687055e2-3e58-11e8-8fed-3acdcb3d2d4c.png)

**Comment by brybrophy** [comment]
I'm not even sure that using the open graph meta tags is within the scope of this project, but it would be an excellent feature to add. If people have taken the time to setup the correct meta tags, the package should be able to use in the popup window.

**Comment by mirik999** [comment]
But, how can i share dynamic articles ? any example ?

**Comment by asiajacko** [comment]
@mirik999 I had the same issue and finally managed to solve it. Can you send me how your Social Media component looks like? In my case this was wrong url pointing to the website general but not /blog. Your meta helmet looks fine.

**Comment by mirik999** [comment]
@asiajacko Social Media component ? you mean js file ?

**Comment by asiajacko** [comment]
@mirik999 yes! js where u have your social media :) @brybrophy exactly. The only way how to test it is: https://developers.facebook.com/tools/debug/sharing/ (and production url) and https://cards-dev.twitter.com/validator for twitter. Using fb debugger you can see exactly what data fb can render, what is missing etc.

**Comment by mirik999** [comment]
@asiajacko https://jsfiddle.net/Mirik999/mhxgkr6v/1/ , it is just react component where setted meta tags with url and thumbnail and with other settings.  and this component where social media button is setted https://jsfiddle.net/Mirik999/ae7mq0gp/

**Comment by asiajacko** [comment]
@mirik999 did u tried : https://developers.facebook.com/tools/debug/sharing/ (and production url) and u can see  exactly what our scraper sees for your URL, where are mistakes and also you can debug there
-- | --

**Comment by SlavikMelnyk** [comment]
It seems like you need server side rendering to make crawlers(fb, tw, etc) see your open graph tags. Dynamically adding on client side does not work. Crawlers can see only static html.

**Comment by brybrophy** [comment]
@SlavikMelnyk That is correct. If you don't want to setup SSR though, you can use a service like prerender.io. This service will cache the html output of your rendered app, and serve the cached version to crawlers and social sharing. The cache updates every 24 hours. It's a great, lightweight solution, and is free up to 250 cached pages.

**Comment by mirik999** [comment]
free up to 250 cached pages -> can u explain in more detail?

**Comment by brybrophy** [comment]
prerender.io is a service the does premiering for you. You can use it for free until you have more than 250 pages cached, then you have to start paying. 

For example, if you have and app that has 10 pages, and and app that has 40 pages, you’ll have 50 pages being cached by prerender.io. These cached pages will be served when your app is being crawled by google, or when a link is shared on social media or texted. 

You don’t need to set up SSR in your app. pretender.io will just render the html output of your app, and keep that around for crawlers. Regular users will be served your actual app. The cached pages are updated every 24 hours.

**Comment by MatteoGioioso** [comment]
So I suppose approach is not working?

**Comment by DeepakKapiswe** [comment]
Does react-share and react-helmet works together for facebook sharing with image thumbnail or not for client rendered web applications ?
